index.js

```diff
@@ -52,10 +61,162 @@ const config = {
+                checksVoidReturn: {
+                    arguments: false,
+                    attributes: false,
+                },
+            },
+        ],
+        "@typescript-eslint/no-unnecessary-type-arguments": "error",
+        "@typescript-eslint/naming-convention": [
+            "warn",
+            {
+                selector: "default",
+                format: ["camelCase"],
+                leadingUnderscore: "allow",
+            },
+            {
+                selector: "variable",
+                format: ["camelCase", "UPPER_CASE"],
+                leadingUnderscore: "allow",
+            },
+            {
+                selector: "parameter",
+                format: ["camelCase"],
+                leadingUnderscore: "allow",
+            },
+            {
+                selector: "property",
+                format: null,
+                leadingUnderscore: "allow",
+            },
+            {
+                selector: "typeLike",
+                format: ["PascalCase"],
+            },
+        ],
+        "@typescript-eslint/padding-line-between-statements": [
+            "warn",
+            {
+                blankLine: "always",
+                prev: "*",
+                next: [
+                    "function",
+                    "interface",
+                    "type",
+                    "try",
+                    "throw",
+                    "case",
+                    "default",
+                ],
+            },
+        ],
+        "import/no-extraneous-dependencies": [
+            "error",
+            // allow devDependencies to be imported into testing files, etc.
+            { devDependencies: ["**/*.{test,spec,story,stories}.{ts,tsx}"] },
+        ],
+        "import/no-default-export": "error",
+        "simple-import-sort/imports": "warn",
+        "simple-import-sort/exports": "warn",
+        "jsdoc/require-throws": "error",
+        "jsdoc/check-indentation": "warn",
+        "jsdoc/no-blank-blocks": "warn",
+        "jsdoc/require-asterisk-prefix": "warn",
+        "jsdoc/require-description": "warn",
+        "jsdoc/sort-tags": "warn",
+        "jsdoc/check-syntax": "warn",
+        "jsdoc/tag-lines": ["warn", "never", { startLines: 1 }],
+        "jsdoc/require-param": ["warn", { checkDestructuredRoots: false }],
+        "jsdoc/require-jsdoc": [
+            "warn",
+            {
+                publicOnly: true,
+                require: {
+                    FunctionDeclaration: true,
+                    FunctionExpression: true,
+                    ArrowFunctionExpression: true,
+                    ClassDeclaration: true,
+                    ClassExpression: true,
+                    MethodDefinition: true,
+                },
+                contexts: [
+                    "VariableDeclaration",
+                    "TSTypeAliasDeclaration",
+                    // Encourage documenting React prop types
+                    "TSPropertySignature",
```

Comment:
It'd be awesome if there was a way to require all React prop types to be extracted into a type and documented. Unfortunately this rule is only triggered when those types are exported directly.